### DISCUSSION TITLE
Question about MQTT Command vs State Topics

### DISCUSSION THREAD

**Discussion by Hect0rG** [tagyoureit/nodejs-poolController#568] [Q&A]
I am working on finally integrating my poolController with OpenHAB and have a question. My understanding of MQTT is that you have the concept of a command topic (used to send commands) and a subscribe topic ( used for reading values). I am trying to figure out what the command topic should be. 

Let's say I am trying to set the RPM's on my pump... What would be my command Topic if this is my state topic (stateTopic="nixie-singlebody/state/pumps/50/pentairvssvrs/rpm)"?

The only other time I have used MQTT is with Tasmota but that did have specific cmd and set commands, I imagine the same applies here.

**Comment by rstrouse** [comment]
Pump RPMs are not set by changing the RPM directly.  The pump RPM is determined by the current demand based upon the currently running circuits and features.  For instance given the setup below, when just the pool is on it will run at 1950 rpm but if the pool and the cleaner are on it will run at 2650.  If by chance the solar kicks on it will increase the rpm to 2750.
![image](https://user-images.githubusercontent.com/47839015/184398831-9fca48ec-0242-41f4-ba28-d64fb18f4a3d.png)

We don't have a single endpoint through MQTT.  However, you can add any interface you like.  There is also a full socket interface as well as a REST command interface that can communicate directly to njsPC.  This eliminates the MQTT broker and will prove to be more responsive.  This is the same interface that dashPanel communicates on as well as a number of other HA interfaces.

**Reply by Hect0rG** [reply]
Ok, that makes sense about the RPM. I'll have to read up on using the socket interface or  REST as I have never used those to be honest. 
Now that being said for a quick solution, would the circuits be able to be turned on via MQTT? I am not trying to avoid using the socket or REST but curious if anything at all can be controlled via MQTT or if it is specifically set up to just reading of values.

**Comment by rstrouse** [comment]
To turn on or off the pool circuit set a topic with the json value like the following.
`nixie-singlebody/state/circuit/setstate` payload `{ "id":6, "state": true | false }`

Use replace circuit with the feature, lightgroup, or circuitgroup for any switchable item.

Then use these topics for some of the other stuff.
`nixie-singlebody/state/heatmode` payload `{ "id": 1, "mode": "off" | "heater" | "solar" | "solarpref" }`
`nixie-singlebody/state/chlorinator` payload `{ "id": 1, "poolSetpoint": 0-100, "superChlor": true | false }`

The rest are in the interfaces wiki.
https://github.com/tagyoureit/nodejs-poolController/wiki/Bindings-Integrations-in-2.0#set-values-with-mqtt
